test/sequential/bootstrap.updatePromotion.test.ts

```diff
@@ -0,0 +1,261 @@
+
+async function changePromoteBootstrapPool(userName: User) {
+  const api = getApi();
+  let result: MangataGenericEvent[];
+
+  const currentPromotingState =
+    await api.query.bootstrap.promoteBootstrapPool();
+
+  if (currentPromotingState) {
+    result = await updatePromoteBootstrapPool(userName, false);
+  } else {
+    result = await updatePromoteBootstrapPool(userName, true);
+  }
+
+  return result;
+}
+
+async function checkBootstrapEvent(checkingEvent: MangataGenericEvent[]) {
+  const filterBootstrapEvent = checkingEvent.filter(
+    (extrinsicResult) => extrinsicResult.method === "Sudid"
+  );
+
+  const userAssetCall = filterBootstrapEvent[0].event.data[0].toString();
+
+  expect(userAssetCall).toContain("Ok");
+}
+
+async function checkBootstrapError(
+  checkingEvent: MangataGenericEvent[],
+  expectedError: string
+) {
+  const api = getApi();
+
+  const filterBootstrapEvent = checkingEvent.filter(
+    (extrinsicResult) => extrinsicResult.method === "Sudid"
+  );
+
+  const userBootstrapErr = hexToU8a(
+    //@ts-ignore
+    filterBootstrapEvent[0].event.data[0].asErr.value.error.toString()
+  );
+
+  const userBootstrapIndex =
+    //@ts-ignore
+    filterBootstrapEvent[0].event.data[0].asErr.value.index.toString();
+
+  const userAssetMetaError = api?.registry.findMetaError({
+    error: userBootstrapErr,
+    index: new BN(userBootstrapIndex),
+  });
+
+  expect(userAssetMetaError.method).toContain(expectedError);
+}
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+
+  keyring = new Keyring({ type: "sr25519" });
+
+  sudo = new User(keyring, sudoUserName);
+});
+
+beforeEach(async () => {
+  const api = getApi();
+  await setupApi();
+
+  testUser1 = new User(keyring);
+
+  [testUser1] = setupUsers();
+
+  const bootstrapCurrencyIssue = await sudoIssueAsset(
+    sudo.keyRingPair,
+    toBN("1", 20),
+    sudo.keyRingPair.address
+  );
+  const bootstrapEventResult = await getEventResultFromMangataTx(
+    bootstrapCurrencyIssue,
+    ["tokens", "Issued", sudo.keyRingPair.address]
+  );
+  const bootstrapAssetId = bootstrapEventResult.data[0].split(",").join("");
+  bootstrapCurrency = new BN(bootstrapAssetId);
+
+  // check that system is ready to bootstrap
+  bootstrapPhase = await api.query.bootstrap.phase();
+  if (bootstrapPhase.toString() === "Finished") {
+    const bootstrapFinalize = await finalizeBootstrap(sudo);
+    eventResponse = getEventResultFromMangataTx(bootstrapFinalize);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  }
+
+  bootstrapPhase = await api.query.bootstrap.phase();
+  expect(bootstrapPhase.toString()).toEqual("BeforeStart");
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(bootstrapCurrency, testUser1),
+    Assets.mintToken(bootstrapCurrency, sudo),
+    Assets.mintNative(testUser1)
+  );
+});
+
+test("bootstrap - Check that we can change promotion bootstrap on each stage before finish", async () => {
```

Comment:
can you double-check this with Shoeb or Mariusz? I think that this behavior is right. the 300 blocks rule should apply AFIK.

Reply:
@AlexChetverov  plz ^^